dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/AwsClient.scala

```diff
@@ -94,6 +97,31 @@ private[dynamodb] trait AwsClient[S <: AwsClientSettings] {
       .withAttributes(ActorAttributes.supervisionStrategy(decider))
       .map(_.asInstanceOf[Op#B])
 
+  def flowOp[Op <: AwsOp](op: Op): Flow[Op, Op#B, NotUsed] = {
+    val opFlow =
+      Flow[Op]
+        .map(op => toAwsRequest(op))
+        .via(connection)
+        .mapAsync(settings.parallelism) {
+          case (Success(response), awsRequestMeta) => toAwsResult(response, awsRequestMeta)
+          case (Failure(ex), _) => Future.failed(ex)
+        }
+
+    opFlow
+      .retryWithBackoff(
+        settings.retrySettings.maximumAttempts,
+        settings.retrySettings.initialRetryTimeout,
+        settings.retrySettings.backoffStrategy, {
+          case _ @(_: InternalServerErrorException | _: ItemCollectionSizeLimitExceededException |
+              _: LimitExceededException | _: ProvisionedThroughputExceededException |
+              _: RequestLimitExceededException) =>
+            Source.single(op).via(opFlow)
```

Comment:
I think we need to improve the retry logic here. As it currently stands, if the stage fails, the retry will happen with the `op` that was passed into the `flowOp` factory at the moment o the flow creation. So it could happen that quite a few operations went through that flow, and then one failed. The retry will happen with the operation that was initially given during the creation of the flow instead of the one, that failed.

I think we should adapt DynamoDb connector flows to be composable with a retry flow in akka-stream-contrib: https://github.com/akka/akka-stream-contrib/blob/master/src/main/scala/akka/stream/contrib/Retry.scala#L42 instead.